api/config/routes.rb

```diff
@@ -73,20 +74,23 @@
             resources :offers, only: [:create]
             post '/email_offer', to: 'offers#email_offer'
             post '/ta/offers/:offer_id/respond_to_offer', to: 'offers#respond'
+
+            get "/pb/:offer_id/pdf" => "offers#get_contract_student"
```

Comment:
Is it necessary to have these routes nested inside session?  Since we already have the `offer_id`, we can directly fetch the offers.